resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/k8s/features/KubernetesFeatureConfigStep.scala

```diff
@@ -18,13 +18,17 @@ package org.apache.spark.deploy.k8s.features
 
 import io.fabric8.kubernetes.api.model.HasMetadata
 
+import org.apache.spark.annotation.DeveloperApi
 import org.apache.spark.deploy.k8s.SparkPod
 
 /**
+ * :: DeveloperApi ::
+ *
  * A collection of functions that together represent a "feature" in pods that are launched for
  * Spark drivers and executors.
  */
-private[spark] trait KubernetesFeatureConfigStep {
+@DeveloperApi
```

Comment:
I think we could mark here `Unstable` as discussed in https://github.com/apache/spark/pull/30206/files#r515606647.

Reply:
My bad, I thought I had pushed that change. Fixed now.